src/content/docs/en/core-concepts/astro-syntax.mdx

```diff
@@ -0,0 +1,191 @@
+---
+title: Astro Syntax
+description: An intro to the .astro component syntax.
+i18nReady: false
+---
+
+**If you know HTML, you already know enough to write your first Astro component.**
+
+Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#differences-between-astro-and-jsx), and adds support for including components and JavaScript expressions.
+
+
+## JSX-like Expressions
+
+You can define local JavaScript variables inside of the frontmatter component script between the two code fences (`---`) of an Astro component. You can then inject these variables into the component's HTML template using JSX-like expressions!
+
+:::note[dynamic vs reactive]
+Using this approach, you can include **dynamic** values that are calculated in the frontmatter.  But once included, these values are not **reactive** and will never change. Astro components are templates that only run once, at build time.
```

Comment:
Whatever solution we come up with ("build or request time", "ahead of time", "build time or on the server") for the problem outlined in #2404 should be reflected here.
```suggestion
Using this approach, you can include **dynamic** values that are calculated in the frontmatter.  But once included, these values are not **reactive** and will never change. Astro components are templates that only run at build time or on the server.
```